### PR TITLE
improve the performance of 'fread' for tcp socket

### PR DESCRIPTION
**Pull request by avriltank**
I found the .net recv becomes very slow when the buffer size is big,it mallocs memory for
the byte[] array,but the gc do not free the memory soon

### CODE REVIEW

File: src/Peachpie.Library/FileSystem.cs [hunk 1]
```diff
@@ -615,6 +621,39 @@ public static PhpString fread(Context ctx, PhpResource handle, int length)
...
 
+            //improve the performance of 'fread' for tcp socket
+            //I found the .net recv becomes very slow when the buffer size is big,it malloc memory for
+            //the byte[] array,but the gc dot not free the memory soon
+            if(stream is SocketStream  theStream)
+            {
+                if(RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    var theSocket = theStream.Socket;
+                    if((int)theSocket.ProtocolType==6)//Tcp
```
Comment (jakubmisek) [inline src/Peachpie.Library/FileSystem.cs:632]: there is an enum for TCP

**Comment by jakubmisek** [comment]
we try to avoid "unsafe" code, isn't there an issue on dotnet for this? Can we just make the buffer smaller?

**Comment by avriltank** [comment]
There is no issue,the Socket.Receive method is very slowly when the buffer size is big.

**Comment by avriltank** [comment]
Yes,you are right!there is not a good idea for peachpie code,i try to find another solution.

**Comment by jakubmisek** [comment]
> I test in my project,the unsafe method can improve Almost 2.5x performance

it is possible, but we can't implement it like this. 

I would recommend implementing a C# method somewhere in your project, for example

```c#
PhpString fread_fast(Context ctx, PhpResource handle, int length) { ... }
```

Or we would need to find a "cleaner" workaround, so we can test it and maintain it.

**Comment by avriltank** [comment]
Yes,that is a good idea,thanks for your reply

**Pull request closed without merging**